wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/MessageJsonAdapter.kt

```diff
@@ -40,8 +41,14 @@ internal class MessageJsonAdapter<M : Message<M, B>, B : Message.Builder<M, B>>(
 
   @Throws(IOException::class)
   override fun toJson(out: JsonWriter, message: M?) {
+    val redactedFieldsAdapter = when (out.tag(RedactedTag::class.java)?.enabled) {
+      true -> redactedFieldsAdapter
+      else -> null
+    } as JsonAdapter<Any?>?
+
     out.beginObject()
-    messageAdapter.writeAllFields(message, jsonAdapters) { name, value, jsonAdapter ->
+    messageAdapter.writeAllFields(message, jsonAdapters,
```

Comment:
I really hate this formatting. Any suggestions?

Reply:
Isn't really better but maybe
```kotlin
messageAdapter.writeAllFields(
  message,
  jsonAdapters,
  redactedFieldsAdapter
) { name, value, jsonAdapter ->
}
```